### PR TITLE
Add web-types for vue 2.7, with support of <script setup> and css v-bind features

### PR DESCRIPTION
**Pull request by PiscesXP**
Vue 2.7 backported features of <script setup> and css v-bind, see:
- https://blog.vuejs.org/posts/vue-2-7-beta.html
- https://github.com/vuejs/vue/blob/main/CHANGELOG.md#270-alpha1-2022-05-31

I add web-types.json for 2.7 with these two features.

### CODE REVIEW

File: vuejs/resources/web-types/vue@2.7.0.web-types.json [hunk 1]
```diff
@@ -0,0 +1,994 @@
...
+          "attribute-value": {
+            "kind": "no-value"
+          }
+        }
+      ]
+    },
+    "css": {
+      "pseudo-elements": [
+        {
+          "name": "v-deep",
```
Comment (DerAlbertCom) [inline vuejs/resources/web-types/vue@2.7.0.web-types.json:979]: ::v-deep This is now also deprecated in vue@2.7. Like in was in vue@3.2,    also :deep(child-selector) should be added

Warning is: `[@vue/compiler-sfc] ::v-deep usage as a combinator has been deprecated. Use :deep(<inner-selector>) instead.`

Comment (DerAlbertCom) [reply vuejs/resources/web-types/vue@2.7.0.web-types.json:979]: image. WebStorm (Latest EAP). Shows `unknown pseudo element dee` warning.

Comment (PiscesXP) [reply vuejs/resources/web-types/vue@2.7.0.web-types.json:979]: I didn't find anything about deep in the vue 2 official documents or the changelog.
Instead, I think it's not a feature of vue but the vue-loader, as it's described in https://vue-loader.vuejs.org/guide/scoped-css.html#deep-selectors.

Comment (DerAlbertCom) [reply vuejs/resources/web-types/vue@2.7.0.web-types.json:979]: Yes, it's from the loader, but that's also the case with `v-deep`. I changed my code from v-deep to deep, and it behaves the same.

Comment (DerAlbertCom) [reply vuejs/resources/web-types/vue@2.7.0.web-types.json:979]: The vue@3.2 web types already have deep declared.

Comment (DerAlbertCom) [reply vuejs/resources/web-types/vue@2.7.0.web-types.json:979]: https://github.com/vuejs/rfcs/blob/master/active-rfcs/0023-scoped-styles-changes.md RFC for that.

**Comment by chemic** [comment]
Any news about it? Will it merged into 2022.2.1? 🤓

**Comment by vepanimas** [comment]
Thank you very much for the PR! Actually, it had been merged a long time ago, I just forgot to close the PR 😅.

The related issue https://youtrack.jetbrains.com/issue/WEB-56412/Support-script-setup-in-SFC-for-VueJS-27

**Comment by dsvgl** [comment]
@vepanimas using latest PhpStorm (2022.2.1) and `:deep()` is still reported as Unknown pseudo selector.
Is this supposed to already have been shipped?

**Pull request closed without merging**